Statistics/Histograms/Histogram.cs

```diff
@@ -22,9 +23,17 @@ public class Histogram
         private bool _ConvergedOnMax = false;
         private ConvergenceCriteria _ConvergenceCriteria;
         private bool _minHasNotBeenSet = false;
+        private const string _type = "Histogram";
```

Comment:
For string constants we do all capital letters and "_" between words. ie: MY_EXAMPLE_WORD.
I was going to suggest that you just make it public and get rid of the property.
public const string HISTOGRAM = "Histogram"
But i see that this is part of your new interface. Since the types are all going to be constants, i would just create an enum HistogramType that has all the types defined and then your interface could have HistogramType HistogramType{get;} and then the implementations would just return the correct type.